cordova-plugin-local-notification/cordova-plugin-local-notification.d.ts

```diff
@@ -0,0 +1,177 @@
+   */
+  on(eventType: string, callback?: (notification?: ICordovaPluginLocalNotificationLocalNotification) => void): void;
+  /**
+   * Update a single scheduled local notification
+   */
+  update(notificationData: ICordovaPluginLocalNotificationLocalNotification): void;
+  /**
+   * Update multiple scheduled local notifications
+   */
+  update(notificationData: Array<ICordovaPluginLocalNotificationLocalNotification>): void;
+  /**
+   * Clear a single triggered local notification
+   */
+  clear(id: number, callback?: () => void): void;
+  /**
+   * Clear multiple triggered local notifications
+   */
+  clear(ids: Array<number>, callback?: () => void): void;
+  /**
+   * Clear all triggered local notifications
+   */
+  clearAll(callback?: () => void, scope?: CordovaPluginLocalNotification) : void;
+  /**
+   * Cancel a single scheduled local notification
+   */
+  cancel(id: number, callback?: () => void): void;
+  /**
+   * Cancel multiple scheduled local notifications
+   */
+  cancel(ids: Array<number>, callback?: () => void): void;
+  /**
+   * Cancel all scheduled local notifications
+   */
+  cancelAll(callback?: () => void, scope?: CordovaPluginLocalNotification): void;
+  /**
+   * Check the local notification's presence
+   */
+  isPresent(id: number, callback: (present: boolean) => void): void;
+  /**
+   * Check if the local notification is scheduled
+   */
+  isScheduled(id: number, callback: (present: boolean) => void): void;
+  /**
+   * Check if the local notification is triggered
+   */
+  isTriggered(id: number, callback: (present: boolean) => void): void;
+  /**
+   * Get the ids of all local notifications
+   */
+  getAllIds(callback: (ids: Array<number>) => void): void;
+  /**
+   * Get the ids of all scheduled local notifications
+   */
+  getScheduledIds(callback: (ids: Array<number>) => void): void;
+  /**
+   * Get the ids of all triggered local notifications
+   */
+  getTriggeredIds(callback: (ids: Array<number>) => void): void;
+  /**
+   * Get a single local notification
+   */
+  get(id: number, callback: (notification: ICordovaPluginLocalNotificationLocalNotification) => void): void;
+  /**
+   * Get multiple local notifications
+   */
+  get(id: Array<number>, callback: (notification: Array<ICordovaPluginLocalNotificationLocalNotification>) => void): void;
+  /**
+   * Get a single scheduled local notification
+   */
+  getScheduled(id: number, callback: (notification: ICordovaPluginLocalNotificationLocalNotification) => void): void;
+  /**
+   * Get a single triggered local notification
+   */
+  getTriggered(id: number, callback: (notification: ICordovaPluginLocalNotificationLocalNotification) => void): void;
+  /**
+   * Get all local notifications
+   */
+  getAll(callback: (notifications: Array<ICordovaPluginLocalNotificationLocalNotification>) => void): void;
+
+}
+
+interface ICordovaPluginLocalNotificationLocalNotification {
```

Comment:
http://www.typescriptlang.org/docs/handbook/writing-declaration-files.html

> Naming Conventions
In general, you shouldn’t prefix interfaces with I (e.g. IColor). Because the concept of an interface in TypeScript is much more broad than in C# or Java, the IFoo naming convention is not broadly useful.